src/audit/secrets_outside_environment.rs

```diff
@@ -0,0 +1,83 @@
+use github_actions_models::{
+    common::{Env, EnvValue, expr::LoE},
+    workflow::job::StepBody,
+};
+
+use super::{Audit, audit_meta};
+use crate::{finding::Confidence, models::StepCommon, AuditState, AuditLoadError, Persona};
+
+pub(crate) struct SecretsOutsideEnvironment;
+
+audit_meta!(
+    SecretsOutsideEnvironment,
+    "secrets-outside-environment",
+    "secrets used without an environment to gate them"
+);
+
+impl Audit for SecretsOutsideEnvironment {
+    fn new(_state: &AuditState<'_>) -> Result<Self, AuditLoadError>
+    where
+        Self: Sized,
+    {
+        Ok(Self)
+    }
+
+    fn audit_step<'w>(
+        &self,
+        step: &crate::models::Step<'w>,
+    ) -> anyhow::Result<Vec<crate::finding::Finding<'w>>> {
+        let mut findings = vec![];
+
+        if step.parent.environment().is_some() {
+            return Ok(findings);
+        }
+
+        let eenv: &Env;
+        match &step.body {
+            StepBody::Uses { uses: _, with } => {
+                eenv = with;
+            }
+            StepBody::Run {
+                run: _,
```

Comment:
Note: I'm okay with skipping `run:` blocks for an MVP here, but we _will_ need to extract expressions from those and scan for secrets as well at some point.

Reply:
Makes sense! I naively thought that wouldn't be required, would this be an example?

```
- name: Clean install dependencies and build
  run: |
    npm ci
    PASSWORD=${{ secrets.password}} npm run build
```